### PR TITLE
FAB-137: Implemented image upload to Azure Blob Storage

### PR DESCRIPTION
**Pull request by SeanMcCrossanTUD**
- Added functionality to generate and upload data profiling visuals to Azure
- Enhanced error handling for scenarios with empty or whitespace-only CSV data
-  Download 'dirty_startup_dataset.csv' when other dataset is blank (Will be removed after Friday)
- Edited DataProfilingVisuals to save visuals as io.Bytes.

### CODE REVIEW

File: python_da/dq_checks/src/app.py [hunk 1]
```diff
@@ -58,17 +60,50 @@ def download_blob_csv_data():
...
+            raise Exception('No blobs found in the container')
+
         logger.info(f'Downloading CSV data from Azure Blob Storage: {blob_name}')
         blob_client = blob_service_client.get_blob_client(container_name, blob_name)
         blob_data = blob_client.download_blob()
         csv_data = blob_data.readall().decode('utf-8')
+
+        if not csv_data.strip(): 
+            logger.warning("Downloaded CSV data is empty or contains only whitespace. Attempting to download 'dirty_startup_dataset.csv' instead.")
+            blob_name = "dirty_startup_dataset.csv"
```
Comment (D22124491-NaveenMaheswaran) [inline python_da/dq_checks/src/app.py:102]: can you declare the variable on top

File: python_da/dq_checks/src/app.py [hunk 2]
```diff
@@ -198,6 +222,43 @@ def upload_results_to_azure(result):
         logger.error(f"Unexpected error while uploading result to Azure Blob Storage: {str(e)}")
         raise e 
 
+def upload_image_to_azure(img_data, blob_name, container_name_images, connection_string):
```
Comment (D22124491-NaveenMaheswaran) [inline python_da/dq_checks/src/app.py:225]: can you create a folder from code, I think if you give container path like ='/contaninername/somestatic value'

**Review by D22124491-NaveenMaheswaran — APPROVED** [review]

File: python_da/dq_checks/src/app.py (hunk 1 above)
Comment (SeanMcCrossanTUD) [reply python_da/dq_checks/src/app.py:102]: Yeah will do, I'll be doing a refactoring next week where I will remove all this stuff. It was mainly for testing on my machine.

File: python_da/dq_checks/src/app.py (hunk 2 above)
Comment (SeanMcCrossanTUD) [reply python_da/dq_checks/src/app.py:225]: Agreed, I'll create jiras I'll add it to the refactoring list. Tbh need to do a greater refactor as python files are getting a bit large. Thinking of adding a lint action to our code next week:
Will help ensure that our code is good quality. 
https://github.com/marketplace/actions/lint-action

File: python_da/dq_checks/src/app.py (hunk 1 above)
Comment (SeanMcCrossanTUD) [reply python_da/dq_checks/src/app.py]: Resolved with latest commit.

**Review by nikodemadamski — APPROVED** [review]
Yes this is perfect, now we have to move onto our 3D visualisation as mentioned in our call. When we meet at 16:00 we must discuss this further. For now your code looks perfect.

**Pull request merged**